lib/poecfg.cpp

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2024
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "poecfg.h"
+#include "tokenize.h"
+
+namespace swss {
+
+std::string PoeConfig::parseKey(const std::string &key_str)
+{
+    std::vector<std::string> token = tokenize(key_str, ':');
+    std::string type, idx;
+
+    if (token.size() == 1)
+    {
+        if (token[0].rfind("Ethernet", 0) != 0)
+        {
+            return "";
+        }
+        type = "port";
+        idx = token.at(0);
+    }
+    else if (token.size() == 2)
+    {
+        type = token.at(0);
+        idx = token.at(1);
+    }
+
```

Comment:
if token.size() is not 1 or 2, this is and error that should be logged and handled.

Reply:
added warning logs